components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/HistoryMetadataSuggestionProvider.kt

```diff
@@ -44,8 +48,16 @@ class HistoryMetadataSuggestionProvider(
             return emptyList()
         }
 
+        val maxReturnedSuggestions = if (maxNumberOfResults > 0) {
+            minOf(maxNumberOfResults, METADATA_SUGGESTION_LIMIT)
+        } else {
+            METADATA_SUGGESTION_LIMIT
+        }
+
         val suggestions = historyStorage
             .queryHistoryMetadata(text, METADATA_SUGGESTION_LIMIT)
+            .filter { it.totalViewTime > 0 }
```

Comment:
This was suggested by Christian, to only returns as results pages that users actually spent time on.